### ISSUE TITLE
enable e10s flag on firefox

### ISSUE THREAD

**Issue by nightpool** [new-xkit/XKit#1263]
Theoretically, we don't use any APIs that should cause an issue here. I tested this for 20 minutes or so on my end and couldn't find any bugs, but would like a second opinion before we flip the switch.

e10s is multiprocess firefox, and addons that don't have the `multiprocess` permission in their manifest file run using shims that slow down both the addon and the browser.

**Comment by nightpool** [comment]
Looks like we may be blocked on this bug though: https://bugzilla.mozilla.org/show_bug.cgi?id=1247879

**Comment by soren121** [comment]
Would it be possible to just port the Chrome extension to Firefox instead? That would automatically get XKit e10s support.

I took a quick look at the Chrome bridge, and Firefox has supported all of the extension API calls used there since v47.

**Comment by hobinjk** [comment]
@soren121 That would be a good approach as long as there is a way to migrate the current storage (in preferences) to the Chrome extension's localStorage.

**Comment by nightpool** [comment]
I mean, to be clear, we have e10s support except for that bug I linked. Can
you guarantee that that bug doesn't apply to the webextension api?

**Comment by soren121** [comment]
@hobinjk Ah, that's a good point. I'm not sure if that's possible, but I could check.

@nightpool The bug report specifically indicates that the bug is in the old Addon SDK and not in the WebExtensions implementation. I could test it to be sure though.

**Comment by soren121** [comment]
Update: it looks like I can migrate the extension storage by utilizing an [Embedded WebExtension](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Embedded_WebExtensions).

I'll experiment with this over the weekend, and if everything goes well I'll open a PR.

And also, yes, I understand that the Firefox extension is e10s ready as-is apart from the linked bug. Given how long the bug has stalled, I don't foresee it getting fixed anytime soon. Transitioning to a WebExtension would simplify maintenance, since the Firefox & Chrome bridges would be very similar, if not the same. Additionally, Mozilla plans to disable all non-WebExtension addons in Firefox 57 (scheduled for November 2017), so...that's certainly an incentive.

**Comment by nar001** [comment]
Hello, just wondering how the WebExtension port is going? Any news?

**Comment by hobinjk** [comment]
I had some time on my hands (by which I mean that I have a thesis due soon and I am procrastiworking).

Anyway, there's a probably-working embedded WebExtension that does migrating on my https://github.com/hobinjk/XKit/tree/firefox-hybrid-migrating-webextension

It needs a lot of testing before I would consider it ready for release (50% chance of not showing up, 45% of not migrating data, 5% chance of just being slow)

**Comment by nightpool** [comment]
This was closed by #1293. It doesn't use a WebExtension, which was postponed to a further release because we're already multiprocess compatible, and just needed to flip the e10s switch.
